### DISCUSSION TITLE
Always panic when the result is null but there's no syntax error

### DISCUSSION THREAD

**Discussion by marcopolooo** [jackc/pgx#1966] [Ideas]
I think we should distinguish between **Syntax Error** or something like that and the result is null but there's no syntax error.
Because we need to add new error handling to be able to differentiate them. Thanks

CMIIW

**Comment by jackc** [comment]
A query that returns no rows is not an error as far as the database is concerned.

But if you are expecting exactly one row then use [QueryRow](https://pkg.go.dev/github.com/jackc/pgx/v5#Conn.QueryRow), [CollectOneRow](https://pkg.go.dev/github.com/jackc/pgx/v5#CollectOneRow), or [CollectExactlyOneRow](https://pkg.go.dev/github.com/jackc/pgx/v5#CollectExactlyOneRow).

If you are expecting at least one row, but consider zero rows an error then you could write a simple wrapper around [CollectRows](https://pkg.go.dev/github.com/jackc/pgx/v5#Conn.QueryRow).
